src/index.js

```diff
@@ -102,39 +132,47 @@ class Dropzone extends React.Component {
 
   onDrop(e) {
     const { onDrop, onDropAccepted, onDropRejected, multiple, disablePreview } = this.props;
-    const fileList = getDataTransferItems(e, multiple);
     const acceptedFiles = [];
     const rejectedFiles = [];
 
     // Stop default browser behavior
     e.preventDefault();
 
-    // Reset the counter along with the drag on a drop.
-    this.enterCounter = 0;
+    // Reset the target list along with the drag on a drop.
+    this.dragTargets = [];
     this.isFileDialogActive = false;
 
-    fileList.forEach((file) => {
-      if (!disablePreview) {
-        file.preview = window.URL.createObjectURL(file); // eslint-disable-line no-param-reassign
-      }
+    // Don't fire callbacks if a child Dropzone component has already handled this event
+    // eslint-disable-next-line no-underscore-dangle
+    if (!e.__dropzoneChildHandledThis) {
+      e.persist();
```

Comment:
Normally event objects that React passes to event handlers are structurally reused, like:

```js
delete e.target;
```

And then put in a pool. If another event comes in, they'll get removed from the pool and sent to another handler.

`e.persist()` will keep that event out of the pool, so we get our own copy of it, and can mutate the event by adding the `__dropzoneChildHandledThis` property. When I tried to do this without an `e.persist()` call, React would throw a warning to the browser console.

Further reading: https://facebook.github.io/react/docs/events.html